cfme/infrastructure/virtual_machines.py

```diff
@@ -86,46 +86,110 @@
 retire_remove_button = "//span[@id='remove_button']/a/img"
 
 match_page = partial(match_location, controller='vm_infra', title='Virtual Machines')
-vm_templates_tree = partial(accordion.tree, "VMs & Templates")
-vms_tree = partial(accordion.tree, "VMs")
-templates_tree = partial(accordion.tree, "Templates")
-
-
-def reset_page():
-    tb.select("Grid View")
-    from cfme.web_ui import paginator
-    if sel.is_displayed(search_box.search_field):
-        search.ensure_normal_search_empty()
-    if paginator.page_controls_exist():
-        # paginator.results_per_page(1000)
-        paginator.check_all()
-        paginator.uncheck_all()
 
 
 drift_table = CheckboxTable("//th[normalize-space(.)='Timestamp']/ancestor::table[1]")
 
 
-class InfraVmDetailsToolbar(View):
-    configuration = Dropdown("Configuration")
-    policy = Dropdown("Policy")
-    lifecycle = Dropdown("Lifecycle")
+class InfraVmDetailsToolbar(VMToolbar):
     monitoring = Dropdown("Monitoring")
-    power = Dropdown("Power")
     access = Dropdown("Access")
+    reload = Button(title='Reload current display')
+
+
+class VmsTemplatesAccordion(View):
+    """
+    The accordion on the Virtual Machines page
+    """
+    @View.nested
+    class vmstemplates(Accordion):  # noqa
+        ACCORDION_NAME = 'VMs & Templates'
+        tree = ManageIQTree()
+
+    @View.nested
+    class vms(Accordion):  # noqa
+        ACCORDION_NAME = 'VMs'
+        tree = ManageIQTree()
+
+    @View.nested
+    class templates(Accordion):  # noqa
+        ACCORDION_NAME = 'Templates'
+        tree = ManageIQTree()
+
 
+class InfraVMView(BaseLoggedInPage):
+    """Base view for header/nav check, inherit for navigatable views"""
 
-class InfraVmDetailsView(BaseLoggedInPage):
-    # TODO this is only minimal implementation for toolbar access through widgetastic
+    including_entities = View.include(VMEntities, use_parent=True)
```

Comment:
You shouldn't include the VMEntities view here, since this view is inherited for both the 'All' page and the 'Details' page below.

I think this class should only have the `in_infra_vms` property